subsys/net/lib/nrf_cloud/src/nrf_cloud_agps.c

```diff
@@ -31,6 +31,7 @@ static K_SEM_DEFINE(agps_injection_active, 1, 1);
 
 static bool agps_print_enabled;
 static struct nrf_modem_gnss_agps_data_frame processed;
+static K_MUTEX_DEFINE(processed_lock);
```

Comment:
The writing should be protected with the mutex as well, right? The current implementation only prevents two reads from happening at the same time, but read could still happen in the middle of writing, which is most likely what we're trying to prevent.

Reply:
I think it would be better to protect the whole A-GPS data processing. With the current implementation, `processed` variable can be read while A-GPS data is being processed (between different data types). It would be better to lock the mutex before A-GPS data processing starts and unlock it when all data has been processed. The code actually has a semaphore called `agps_injection_active` which is used to prevent two simultaneous calls to `nrf_cloud_agps_process()`. Maybe these could be combined. But anyway, the `k_sem_take()` and `k_sem_give()` calls are pretty much where the protected block should start and end, I think.

Reply:
Ok, I updated the implementation already. I see that your suggestion differs a bit from how I did it. I will have to revisit this tomorrow.

Reply:
Alright, I updated the commit, please have a look. The mutex now protects `agps_send_to_modem` which is the only function in addition to `nrf_cloud_agps_processed` that actually reads/writes the `processed` variable.

Reply:
Looks good. There's still the `memset()` in `nrf_cloud_agps_request()` which is not protected, but I'm not sure if it's possible for the thread execution to be interrupted while executing that call. An ISR is at least possible, but we can't block an ISR with a mutex anyway. I'll leave it up to you whether to protect that with the mutex as well.

Reply:
Woops, I missed that one. Re-added it just to be safe.